lib/apiservers/service/restapi/handlers/vch_create_test.go

```diff
@@ -0,0 +1,296 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	cli "gopkg.in/urfave/cli.v1"
+
+	"github.com/vmware/govmomi/list"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/cmd/vic-machine/common"
+	"github.com/vmware/vic/cmd/vic-machine/create"
+	"github.com/vmware/vic/lib/apiservers/service/models"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+type mockFinder struct {
+	path string
+}
+
+func (mf mockFinder) Element(ctx context.Context, t types.ManagedObjectReference) (*list.Element, error) {
+	return &list.Element{
+		Path: t.Value,
+	}, nil
+}
+
+func TestFromManagedObject(t *testing.T) {
+	op := trace.NewOperation(context.Background(), "TestFromManagedObject")
+	var m *models.ManagedObject
+
+	expected := ""
+	actual, err := fromManagedObject(op, nil, "t", m)
+	assert.NoError(t, err, "Expected nil error, got %#v", err)
+	assert.Equal(t, expected, actual, "Expected %s, got %s", expected, actual)
+
+	m = &models.ManagedObject{
+		Name: "testManagedObject",
+	}
+
+	mf := mockFinder{}
+
+	expected = m.Name
+	actual, err = fromManagedObject(op, mf, "t", m)
+	assert.NoError(t, err, "Expected nil error, got %#v", err)
+	assert.Equal(t, expected, actual, "Expected %s, got %s", expected, actual)
+
+	m.ID = "testID"
+
+	expected = m.ID
+	actual, err = fromManagedObject(op, mf, "t", m)
+	assert.NoError(t, err, "Expected nil error, got %#v", err)
+	assert.Equal(t, expected, actual, "Expected %s, got %s", expected, actual)
```

Comment:
Curious (since I'm new to unit testing in go): what's the goal of explicitly passing `Expected %s, got %s", expected, actual`? Shouldn't the default behavior be to print something along those lines?

Reply:
I've been using this pattern since I started because of some other test I referenced in the beginning. :)

But, upon investigating:
```go
func TestTest(t *testing.T) {                
        assert.Equal(t, "expected", "actual")
}
```
```
--- FAIL: TestTest (0.00s)
  Error Trace:  vch_create_test.go:47
  Error:    Not equal: "expected" (expected)
              != "actual" (actual)
```

Reply:
Fixed